### PR TITLE
Validate parameters against the actual compose file

### PR DESCRIPTION
**Pull request by ndeloof**
**- What I did**
add a validation step to ensure parameters are actually used by compose file

**- How I did it**
Reused ExtractVariables from `init` and compare with parameters

**- How to verify it**
validate a dockerapp with unecessart parameters

**- Description for the changelog**
docker app validate do check parameters are in actual use by compose file

**- A picture of a cute animal (not mandatory but encouraged)**
image

### CODE REVIEW

File: internal/commands/validate.go [hunk 1]
```diff
@@ -4,6 +4,9 @@ import (
 	"fmt"
 	"os"
 
+	"github.com/docker/app/internal/compose"
```
Comment (silvin-lubecki) [inline internal/commands/validate.go]: import order looks wrong

File: internal/commands/validate.go [hunk 2]
```diff
@@ -23,22 +26,37 @@ func validateCmd() *cobra.Command {
...
+		return err
+	}
+
+	vars, err := compose.ExtractVariables(app.Composes()[0], compose.ExtrapolationPattern)
+	if err != nil {
+		return errors.Wrap(err, "failed to parse compose file")
+	}
+	for k := range app.Parameters().Flatten() {
+		if _, ok := vars[k]; !ok {
+			return fmt.Errorf("%s is declared as parameter but not used by the compose file", k)
```
Comment (silvin-lubecki) [inline internal/commands/validate.go]: Can we have a test on this feature?

File: internal/commands/validate.go (hunk 1 above)
Comment (ndeloof) [reply internal/commands/validate.go]: ? I ran goimports

Comment (ndeloof) [reply internal/commands/validate.go]: ok so any blank line just prevent goimports to work the way I expected it to. Nice to know :roll_eyes:

File: e2e/validate_test.go [hunk 3]
```diff
@@ -0,0 +1,20 @@
+package e2e
+
+import (
+	"path"
+	"strings"
+	"testing"
+
+	"gotest.tools/assert"
+
```
Comment (silvin-lubecki) [inline e2e/validate_test.go]: import order looks wrong 🤣

File: internal/commands/validate.go (hunk 2 above)
Comment (silvin-lubecki) [reply internal/commands/validate.go]: Sir yes Sir!

**Pull request merged**